packages/server/lib/cloud/api/get_app_studio.ts

```diff
@@ -0,0 +1,126 @@
+const serverFilePath = path.join(studioPath, 'server', 'index.js')
+
+const downloadAppStudioBundleToTempDirectory = async (projectId?: string): Promise<void> => {
+  let responseSignature: string | null = null
+
+  await (asyncRetry(async () => {
+    const response = await fetch(routes.apiRoutes.studio() as string, {
+      // @ts-expect-error - this is supported
+      agent,
+      method: 'GET',
+      headers: {
+        'x-route-version': '1',
+        'x-cypress-signature': cloudApi.publicKeyVersion,
+        ...(projectId ? { 'x-cypress-project-slug': projectId } : {}),
+        'x-cypress-studio-mount-version': '1',
+        'x-os-name': os.platform(),
+        'x-cypress-version': pkg.version,
+      },
+      encrypt: 'signed',
+    })
+
+    responseSignature = response.headers.get('x-cypress-signature')
+
+    await new Promise<void>((resolve, reject) => {
+      const writeStream = fs.createWriteStream(bundlePath)
+
+      writeStream.on('error', reject)
+      writeStream.on('finish', () => {
+        resolve()
+      })
+
+      // @ts-expect-error - this is supported
+      response.body?.pipe(writeStream)
+    })
+  }, {
+    maxAttempts: 3,
+    retryDelay: _delay,
+    shouldRetry: isRetryableError,
+  }))()
+
+  if (!responseSignature) {
+    throw new Error('Unable to get studio signature')
+  }
+
+  const verified = await verifySignatureFromFile(bundlePath, responseSignature)
+
+  if (!verified) {
+    throw new Error('Unable to verify studio signature')
+  }
+}
+
+const getTarHash = async (): Promise<string> => {
+  let hash = ''
+
+  return new Promise<string>((resolve, reject) => {
+    fs.createReadStream(bundlePath)
+    .pipe(crypto.createHash('sha256'))
+    .setEncoding('base64url')
+    .on('data', (data) => {
+      hash += String(data)
+    })
+    .on('error', reject)
+    .on('close', () => {
+      resolve(hash)
+    })
+  })
+}
+
+export const getAppStudio = async (projectId?: string): Promise<StudioManager> => {
+  try {
+    let script: string
+    let studioHash: string | undefined
+
+    // First remove studioPath to ensure we have a clean slate
+    await fs.promises.rm(studioPath, { recursive: true, force: true })
+    await ensureDir(studioPath)
+
+    if (process.env.CYPRESS_LOCAL_STUDIO_PATH) {
+      const appPath = path.join(process.env.CYPRESS_LOCAL_STUDIO_PATH, 'app')
+      const serverPath = path.join(process.env.CYPRESS_LOCAL_STUDIO_PATH, 'server')
+
+      await copy(appPath, path.join(studioPath, 'app'))
+      await copy(serverPath, path.join(studioPath, 'server'))
+    } else {
+      await downloadAppStudioBundleToTempDirectory(projectId)
+
+      studioHash = await getTarHash()
+
+      await tar.extract({
+        file: bundlePath,
+        cwd: studioPath,
+      })
+    }
+
+    script = await readFile(serverFilePath, 'utf8')
+
+    const appStudio = new StudioManager()
+
+    await appStudio.setup({ script, studioPath, studioHash })
+
+    return appStudio
+  } catch (error) {
```

Comment:
should we type the error here since anything can throw and make sure we pass the right type into `createInErrorManager`?

Reply:
Our ts rules require `any` or `unknown` so I'm going to specify `any`.

Reply:
Thoughts on this pattern? [`634109a` (#31033)](https://github.com/cypress-io/cypress/pull/31033/commits/634109a70223aa69af77b1fa5415ce3891556128)

Reply:
I like that pattern!